core/vm/evm.go

```diff
@@ -41,10 +41,10 @@ type (
 func (evm *EVM) precompile(addr common.Address) (PrecompiledContract, bool) {
 	var precompiles map[common.Address]PrecompiledContract
 	switch {
-	case evm.chainRules.IsPrague:
-		precompiles = PrecompiledContractsBerlin
 	case evm.chainRules.IsCancun:
 		precompiles = PrecompiledContractsCancun
+	case evm.chainRules.IsPrague:
+		precompiles = PrecompiledContractsBerlin
```

Comment:
Just to express the correct order since we're rebased on top of shapella, not cancun.